src/app/containers/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,76 @@
+import * as React from 'react';
+import { connect } from "react-redux";
+import { Request, State } from "../../types";
+import { searchRequestSent } from "../../actions";
+import './SearchBar.css';
+
+type SearchBarStateProps = {
+    requestState: Request;
+}
+
+type SearchBarDispatchProps = {
+    onSearchRequestSend: (searchString: string) => void;
+}
+
+type SearchBarProps = SearchBarDispatchProps & SearchBarStateProps;
+
+type SearchBarState = {
+    searchString: string;
+};
+
+export class SearchBar extends React.Component<SearchBarProps, SearchBarState>{
+    state = {
+        searchString: ''
+    };
+
+    handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+        event.preventDefault();
+
+        const searchTerm = this.state.searchString;
+        const { onSearchRequestSend } = this.props;
+        onSearchRequestSend(searchTerm);
+    };
+
+    handleSearchStringChange: React.ChangeEventHandler<HTMLInputElement> = event => {
+        this.setState({
+            ...this.state,
+            searchString: event.target.value
+        })
+    };
+
+    render(){
+        return (
+          <form onSubmit={this.handleSubmit}>
+              <input
+                  type={'search'}
+                  name=''
+                  placeholder={'Enter search string'}
+                  onChange={this.handleSearchStringChange}
+                  value={this.state.searchString}
+              />
+              <button type='submit' >
+                  Search
+              </button>
+          </form>
+        )
+    }
+}
+
+const mapStateToProps = (state: State) => {
```

Comment:
research the type safety on this method of splitting up mapState and mapDispatch versus writing them inline within the connect